firmware/library/L3_HAL/apds_9960.hpp

```diff
@@ -1,20 +1,518 @@
+    {
+      // Check initial point, then check next point
+      // to figure out the direction of the swipe
+      switch (points_[i])
+      {
+        case Direction::kUp:
+          if (points_[i + 1] == Direction::kDown)
+          {
+            result = kSwipeDOWN;
+          }
+          break;
+        case Direction::kDown:
+          if (points_[i + 1] == Direction::kUp)
+          {
+            result = kSwipeUP;
+          }
+          break;
+        case Direction::kLeft:
+          if (points_[i + 1] == Direction::kRight)
+          {
+            result = kSwipeRIGHT;
+          }
+          break;
+        case Direction::kRight:
+          if (points_[i + 1] == Direction::kLeft)
+          {
+            result = kSwipeLEFT;
+          }
+          break;
+        default: break;
+      }
+    }
+    return result;
+  }
+  Gesture GetGesture() override
+  {
+    if (CheckIfGestureOccured())
+    {
+      //  Check if gesture ended
+      if (ReadGestureMode() == 0)
+      {
+        uint8_t level = 0;
+        level         = (GetGestureFIFOLevel());
+        ReadGestureFIFO(gfifo_data_, level);
+
+        // reset processing variables
+        index_      = 0;
+        near_count_ = 0;
+        far_count_  = 0;
+        for (int i = 0; i < 8; i++)
+        {
+          points_[i] = Direction::kNone;
+        }
+
+        ProcessGestureData(level);
+
+        return DecodeGestureData(level);
+      }
+    }
+    return Gesture::kERROR;
+  }
+  constexpr Apds9960()
+      // sensitivities used to calibrate cross-talk of IR light
+      : up_sensitivity_(0),
+        down_sensitivity_(0),
+        left_sensitivity_(0),
+        right_sensitivity_(0),
+        far_sensitivity_(750),
+        near_sensitivity_(1000),
+        far_count_(0),
+        near_count_(0),
+        index_(0)
+  {
+    for (int i = 0; i < kMaxDataSize; i++)
+    {
+      points_[i] = Direction::kNone;
+    }
+    for (int i = 0; i < kMaxFifoSize; i++)
+    {
+      gfifo_data_[i] = 0;
+    }
+  }
+  constexpr Apds9960(int8_t up_sensitivity, int8_t down_sensitivity,
+                     int8_t left_sensitivity, int8_t right_sensitivity,
+                     int8_t far_sensitivity, int8_t near_sensitivity)
+      : up_sensitivity_(up_sensitivity),
+        down_sensitivity_(down_sensitivity),
+        left_sensitivity_(left_sensitivity),
+        right_sensitivity_(right_sensitivity),
+        far_sensitivity_(far_sensitivity),
+        near_sensitivity_(near_sensitivity),
+        far_count_(0),
+        near_count_(0),
+        index_(0)
+  {
+    for (int i = 0; i < kMaxDataSize; i++)
+    {
+      points_[i] = Direction::kNone;
+    }
+    for (int i = 0; i < kMaxFifoSize; i++)
+    {
+      gfifo_data_[i] = 0;
+    }
+  }
+  ~Apds9960() {}
+
+ private:
+  // Max FIFO size is represented by the max 4x32 FIFO for
+  // Up/Down/Left/Right gesture data
+  static constexpr uint8_t kMaxFifoSize = 128;
+  static constexpr uint8_t kMaxDataSize = 8;
```

Comment:
Do you think the user would ever want or need to change these values? If so, we should put this in the config.hpp file. If not, then put a comment saying to keep these values as they are and where you found/calculated them.